### ISSUE TITLE
Promise for post patch  and delete

### ISSUE THREAD

**Issue by mihaisc** [devnixs/ODataAngularResources#115]
Hello,

I am using this library with an web api 2 odata server. I have a simple use case : a table of items with add, edit, delete buttons. When i add an item a modal opens. After i add the item i want the modal to close and the table to refresh. My problem is that the modal closes and the table refreshes after the item is added, because i can't wait for the $save() method. 

This is the method in my service : 

```
function _addDocument(data) {
            var Doc = $odataresource(urlConstants.docs, {}, {}, { odatakey: 'Id', isodatav4: true });
            var myDoc = new Doc();
            myDoc.Data = data.Data;
            myDoc.IdCompanie = data.IdCompanie;
            myDoc.Observatii = data.Observatii;
            myDoc.IdTipDosar = data.IdTipDosar;
            myDoc.TipDocument = data.TipDocument
            myDoc.UrlDocuware = data.UrlDocuware;
            myDoc.$save();  
            return myDoc; }

```

And in my controller :

```
 var ress = restServices.addDocument(body);
 $uibModalInstance.close();
```

So basically i have to close the modal after i get the response from $save().

Thank you

**Comment by mihaisc** [comment]
I've managed to find an answer, and i strongly think this should be in the documentation.

```
    ress.$save(function () {
                $uibModalInstance.close();
            });
```
Thank you

**Comment by subpx** [comment]
https://docs.angularjs.org/api/ngResource/service/$resource

**Comment by kyse** [comment]
Agreed, the library is based on angular resources.  But it never hurts to be extra thorough with documentation.
